x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/utils.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import seedrandom from 'seedrandom';
+
+const DEFAULT_PARTIAL_SHUFFLE_SEED = 'seed';
+
+/**
+ * Partially shuffles an array using the Fisher-Yates algorithm.
+ *
+ * The array is shuffled in place, so that:
+ *   - the first `start` elements are kept in place;
+ *   - the elements in the slice from start to end represent the random sample;.
+ *   - the order of elements after end can be arbitrary (but the same over invocations).
+ *
+ * The result is reproducible for the given random seed.
+ *
+ * Examples:
+ *   - shuffle the whole array: partialShuffleArray(arr)
+ *   - shuffle the first 5 elements: partialShuffleArray(arr, 0, 5)
+ *   - keep the first element, shuffle the rest: partialShuffleArray(arr,1)
+ *   - shuffle the last 5 elements: partialShuffleArray(arr, arr.length - 5)
+ *
+ * @param arr - The array to be partially shuffled.
```

Comment:
For partial shuffle, does that mean it will shuffle the whole array, but only pick X elements from it afterwards? Or do you pick X elements and just shuffle them? Because we did want to shuffle all of them so we pick random samples from the provided logfile rather than just shuffling the first X elements

Reply:
No, we still pick from all of the elements after `start`. I've updated the docs so hopefully this is clear now.

Reply:
Gotcha, at some point when I was reading it, it looked like you picked the first X elements and only shuffled those, as that would not have any impact. But if its picking elements at random after start, that's all good :)

Reply:
Well the tests also show some examples